README.md

```diff
@@ -4,3 +4,96 @@
+This mapping is done using a set of predefined patterns and kafka topic templates. The MQTT Bridge uses these patterns to map MQTT topics to Kafka topics. 
+As a part of the bridge, a Kafka producer will be responsible for producing messages from the MQTT clients to the Kafka Cluster.
+
+#### 1.2. Topic Mapping Rules(TOMAR)
+
+The TOMAR is a set of patterns the user provides defining how the MQTT Bridge maps MQTT topic names to Kafka topic names.
+A Mapping Rule is a model that contains an `MQTT topic pattern` and a `Kafka topic template`. 
+All the incoming MQTT message's topic should match an `MQTT topic pattern` in the `TOMAR` so that the bridge knows in which Kafka topic to produce this message. This Kafka topic is defined by a template, `kafka Topic template`. However, if the incoming MQTT message's topic does not match any pattern in the `TOMAR`, the Bridge has a default Kafka topic where the incoming message will be mapped to, know as `messages_default`.
+
+A valid TOMAR is a JSON file that contains an array of mapping rules. Each mapping rule is a JSON object that contains two properties: `mqttTopic` and `kafkaTopic`. The following is an example of a valid TOMAR:
+
+```json
+[
+  {
+    "mqttTopic": "building/{building}/room/{room}/#",
+    "kafkaTopic": "building_{building}_room_{room}"
+  },
+  {
+    "mqttTopic": "sensors/#",
+    "kafkaTopic": "sensor_others"
+  },
+  {
+    "mqttTopic": "sensors/+/data",
+    "kafkaTopic": "sensor_data"
+  }
+]
+```
+
+The wildcard "#" represents one or more levels in the MQTT topic hierarchy. The wildcard "+" represents a single level in the MQTT topic hierarchy.
+It worth's mentioning that it is the user's responsibility to adhere to the [MQTT 3.1.1 naming conventions](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718106) when defining the MQTT topic patterns.
+
+The parts with curly braces in the MQTT topic pattern are called `placeholders`. The MQTT Bridge will extract the values of these placeholders from the MQTT topic and use them to construct the Kafka topic name using the `kafkaTopic` template.
+
+Let's go through each rule to understand how the MQTT Bridge uses these rules to map MQTT topics to Kafka topics:
+
+1. MQTT Topic: "building/{building}/room/{room}/#" -> Kafka Topic: "building_{building}room{room}"
+
+    This rule maps MQTT topics of the form "building/{building}/room/{room}/#" to the Kafka topic "building_{building}room{room}". For example, if the MQTT topic is "building/A/room/1/floor/2", it will be mapped to the Kafka topic "building_A_room_1_floor_2".
+
+2. MQTT Topic: "sensors/#" -> Kafka Topic: "sensor_others"
+
+    This rule maps any MQTT topic starting with "sensors/" followed by any number of levels in the hierarchy to the Kafka topic "sensor_others". For example, if the MQTT topic is "sensors/temperature/living-room",  it will be mapped to the Kafka topic "sensor_others".
```

Comment:
```suggestion
    This rule maps any MQTT topic starting with `sensors/` followed by any number of levels in the hierarchy to the Kafka topic `sensor_others`. 
    For example, if the MQTT topic is `sensors/temperature/living-room`,  it will be mapped to the Kafka topic `sensor_others`.
```

one sentence, one line